### ISSUE TITLE
Method getDateModified of RSS reader doesn't iterate over different formats

### ISSUE THREAD

**Issue by Klaasie** [laminas/laminas-feed#32]
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 2.13.0

#### Summary

The current code suggest that when retrieving the date modified variable it will attempt to create it from 3 different formats: `RSS`, `RFC822` and `RFC2822`. This doesn't happen because the method used `date_create_from_format` doesn't throw an exception.
This also means that if the format is not recognised at all the exception won't be thrown.

#### Current behavior

If the given date doesn't match the first format tested (`RSS`) the variable will resolve into false and the loop will break there.

#### How to reproduce

Read an RSS feed with a date modified format different from `RSS` (`D, d M Y H:i:s O`).

#### Expected behavior

The code will iterate over all 3 different formats and if unable to create the datetime object it will throw an exception saying the date format was not recognised.

For reference, the piece of code with the bug:
https://github.com/laminas/laminas-feed/blob/2.13.x/src/Reader/Feed/Rss.php#L205

```
$dateStandards = [
    DateTime::RSS,
    DateTime::RFC822,
    DateTime::RFC2822,
    null,
];

foreach ($dateStandards as $standard) {
    try {
        $date = date_create_from_format($standard, $dateModified);
        break;
    } catch (\Exception $e) {
        if ($standard === null) {
            throw new Exception\RuntimeException(
                'Could not load date due to unrecognised format'
                . ' (should follow RFC 822 or 2822): ' . $e->getMessage(),
                0,
                $e
            );
        }
    }
}
```

**Comment by froschdesign** [comment]
@Klaasie 
Thanks for the issue report, I will check the problem.

**Comment by weierophinney** [comment]
Fixed with #33.
